### PR TITLE
Fix GitHub CI

### LINKED ISSUE #181: [TwitchAlert] Add timeout to API requests
### Describe the bug
Occasionally GH actions will not successfully run tests (likely something to do with API restrictions, an issues itself). However in this case TwitchAlert tests will continually try to get a response from the API an not stop.

### To Reproduce
Numbered steps to reproduce the behavior:
1. Run GitHub Actions tests
2. Get lucky (idk how to reproduce)
3. Watch the tests continue forever

### Expected behaviour
Whenever requesting from the Twitch API fails, timeout for random time.
If timed out 5 times, throw an error.

### Additional context
- Version e.g. 0.4.2

### PR DESCRIPTION
**Pull request by JayDwee**
Added timeout to TwitchAlert requests_get and adjusted GH actions CI

Close #181

### CODE REVIEW

File: cogs/TwitchAlert.py [hunk 1]
```diff
@@ -693,8 +699,9 @@ async def requests_get(self, url, headers=None, params=None):
 
                 if response.status == 401:
```
Comment (VirajShah18) [inline cogs/TwitchAlert.py:700]: Is a 401 error given when we've reached the rate limit?
And do we not want to retry for different error codes and not just a 401? E.g. a 503 or 504?

File: cogs/TwitchAlert.py [hunk 2]
```diff
@@ -749,8 +756,7 @@ async def get_team_users(self, team_id):
         :return: the JSON information of the users
         """
         url = 'https://api.twitch.tv/helix/teams?name=' + team_id
-        return (
-            await self.requests_get(url)).get("data")[0].get("users")
+        return (await self.requests_get(url)).get("data")[0].get("users")
```
Comment (VirajShah18) [inline cogs/TwitchAlert.py:759]: If using `requests_get()` like this then we assume that "data" and "users" will always be in the response. However this is not guaranteed by `requests_get()` as you don't validate that a correct response has been received. Only a 401 error is actually handled in `requests_get()`

**Review by VirajShah18 — CHANGES_REQUESTED** [review]

File: cogs/TwitchAlert.py (hunk 1 above)
Comment (VirajShah18) [reply cogs/TwitchAlert.py:700]: Performing it recursively like we do now but adding other error codes wouldn't result in an infinite loop. E.g. any request can have a max of 5 failed attempts before aborting no matter the error code returned.

Either way we don't handle other errors at all in this implementation.

Comment (VirajShah18) [reply cogs/TwitchAlert.py:700]: And for certain errors we can simply abort after 1 error code is recieved but we should definitely be throwing an error if there isn't a valid response.

Comment (JayDwee) [reply cogs/TwitchAlert.py:700]: tbh all of this is working atm and is unrelated to this PR for fixing GitHub CI issues, please can they be made into a separate issue, or added to #105 so I can work on it after exams (I am already planning on remaking large parts of TwitchAlert)

File: cogs/TwitchAlert.py (hunk 2 above)
Comment (JayDwee) [reply cogs/TwitchAlert.py:759]: This issue seems to be in multiple places so I'll have to take more time changing things, yet again something to be added to another issue as unrelated from this pull request (only change was to formatting). After the 3rd I should be able to rewrite TwitchAlert to fix all of these issues

File: cogs/TwitchAlert.py (hunk 1 above)
Comment (JayDwee) [reply cogs/TwitchAlert.py:700]: I think I'm going to move to using [pytwitchapi](https://pytwitchapi.readthedocs.io/en/latest/index.html) or a similar library so I don't have to maintain rate limiting etc.

Comment (JayDwee) [reply cogs/TwitchAlert.py:700]: I have updated #105 to reflect these issues

**Review by VirajShah18 — APPROVED** [review]

**Pull request merged**